cpp/monitoring/gcm/exporter.cc

```diff
@@ -0,0 +1,264 @@
+#include "monitoring/gcm/exporter.h"
+
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+#include <sstream>
+
+#include "monitoring/metric.h"
+#include "monitoring/registry.h"
+#include "net/url.h"
+#include "util/json_wrapper.h"
+#include "util/sync_task.h"
+
+DEFINE_string(google_compute_monitoring_push_url, "",
+              "GCM custom timeseries write URL.");
+DEFINE_string(google_compute_monitoring_metrics_url, "",
+              "GCM custom timeseries metrics URL.");
+DEFINE_int32(google_compute_monitoring_push_interval_seconds, 5,
+             "Seconds between pushing metric values to GCM.");
+DEFINE_string(google_compute_metadata_url,
+              "http://metadata/computeMetadata/v1/instance/service-accounts",
+              "URL of GCE metadata server.");
+DEFINE_string(google_compute_monitoring_service_account, "default",
+              "Which GCE service account to use for pushing metrics.");
+
+namespace cert_trans {
+
+using std::chrono::seconds;
+using std::chrono::system_clock;
+using std::chrono::minutes;
+using std::make_pair;
+using std::mutex;
+using std::ostringstream;
+using std::placeholders::_1;
+using std::string;
+using std::thread;
+using std::unique_lock;
+using util::Executor;
+using util::SyncTask;
+
+
+namespace {
+
+
+// Label and Metric -name prefix.
+const char kCloudPrefix[] = "custom.cloudmonitoring.googleapis.com/ct/";
```

Comment:
Is this a landgrab for /ct, or is it scoped to within our monitoring app?  If it's scoped to within our app, why do you need the custom.blah.blah.com/ prefix?